### ISSUE TITLE
Maybe a optimization for vminsert.

### ISSUE THREAD

**Issue by changshun-shi** [VictoriaMetrics/VictoriaMetrics#5440]
### Is your question request related to a specific component?

Yes, vminsert in the cluster version.

### Describe the question in detail

I find some warn logs in vminsert. which shows that `maxLabelsPerTimeseries` is too small, so it dropped the label `vmange`.
However `vmrange` is the official label for `VictoriaMetrics Histogram`.
Maybe when check labels count, exclude vmrange is better.

**Comment by hagen1778** [comment]
Hi @changshun-shi! Thanks for report!

> Maybe when check labels count, exclude vmrange is better.

I don't think it is worth it. Dropping labels is already an event that requires a quick response from the user, it is not something user should allow to happen. Even if we keep `vmrange` label, the dropped label could be the one which differentiates one time series from another. When this happens, database starts storing duplicates with different values - which is much worse as may lead to unpredictable query restuls.

tl;dr; label drop events need to be mitigated ASAP by either fixing the source of the issue or bumping the limits. It doesn't matter which labels were dropped, as there are no guarantees that time series remain unique after such drops.

**Comment by changshun-shi** [comment]
I'm sorry that I don't think so.
If we use **VictoriaMetrics-histogram** in our system, we must set `maxLabelsPerTimeseries` more than 488(486 for bucket and 1 for sum and 1 for count), it's absolutely OK for **VictoriaMetrics-histogram** but 488 is to large for other `metrics`. It's not recommended to use **VictoriaMetrics-histogram** in small `metric system` ?

**Comment by hagen1778** [comment]
>we must set maxLabelsPerTimeseries more than 488(486 for bucket and 1 for sum and 1 for count)

I do not follow. `maxLabelsPerTimeseries` is limiting number of labels per each unique time series individually.
For example, [vm_index_search_duration_seconds_bucket{pod="vmstorage-benchmark-vm-cluster-0", instance="10.71.0.11:8482"}](https://play.victoriametrics.com/select/accounting/1/6a716b0f-38bc-4856-90ce-448fd713e3fe/prometheus/graph/#/?g0.expr=count%28vm_index_search_duration_seconds_bucket%7Bpod%3D%22vmstorage-benchmark-vm-cluster-0%22%2C+instance%3D%2210.71.0.11%3A8482%22%7D%29&g0.range_input=30m&g0.end_input=2023-12-13T15%3A48%3A45&g0.tab=1&g0.relative_time=last_30_minutes&g1.expr=vm_index_search_duration_seconds_bucket%7Bpod%3D%22vmstorage-benchmark-vm-cluster-0%22%2C+instance%3D%2210.71.0.11%3A8482%22%7D&g1.range_input=30m&g1.end_input=2023-12-13T15%3A48%3A45&g1.tab=1&g1.relative_time=last_30_minutes) has 100 time series, each time series has 7 labels. The `maxLabelsPerTimeseries` is applied to check whether 7 doesn't exceed the value. 
I do not understand why you need to have maxLabelsPerTimeseries=488.

**Comment by changshun-shi** [comment]
So sorry that I misunderstood the means of this params. 
I'll close this issue. 
Thanks for your help.
